src/core/Akka/IO/ByteBuffer.cs

```diff
@@ -0,0 +1,115 @@
+//-----------------------------------------------------------------------
+// <copyright file="ActorBase.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+//     Copyright (C) 2013-2015 Akka.NET project <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+
+namespace Akka.IO
+{
+    public class ByteBuffer
```

Comment:
@Aaronontheweb how does it compare to buffer pools in Helios? Should we stay with that or maybe we could reuse Helios implementation?